### PR TITLE
Q2 rebuild and touch-up

### PR DESCRIPTION
**Pull request by tessam30**
When you have a chance, review and see if you can reproduce on your machine. We need a country to run through to do a validation check on.

### CODE REVIEW

**Comment by achafetz** [comment]
FYI  for checking out/reviewing a pull locally (will check later this morning)
image

**Comment by achafetz** [comment]
commented above and made some edits that should have been pushed into the scripts on this branch.

File: Scripts/01_assemble_data_Q2.R [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+# USING TABLEAU DATA + SERVER EXTRACTS
+    
+  # Load data from most recent to least recent; Filtering as needed
+    hfr_pd678 <- vroom(here(datim_folder, "Viewforperiods678_06112020.csv"))
+    
+    hfr_pd5 <- vroom(here(datim_folder, "Viewforperiods567_05262020.csv")) %>% 
+      filter(hfr_pd == 5)
+
+    hfr_pd4 <- vroom(here(datim_folder, "Viewforperiods456_05262020.csv")) %>% 
+      filter(hfr_pd == 4)
```
[the remark this answers is not in the document]
Comment (tessam30) [reply Scripts/01_assemble_data_Q2.R]: Could be given the date stamp is the same on the views.

File: Scripts/01_assemble_data_Q2.R [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  #identify dates to repeat df_datim
+    dates <- as_date(start_date) %>% seq(by = 7, length.out = weeks)
+  
+  #repeat df_datim so there is a full target/result set for each week
+    df_datim_rpt <- map_dfr(.x = dates,
+                            .f = ~mutate(df_datim, date = .x))
+  
+  #remove files (just keep zipped folder)
+    #unlink(files)
+    list.files(datim_folder, "3.1 DATIM", full.names = T) %>% unlink(., recursive = TRUE)
```
[the remark this answers is not in the document]
Comment (tessam30) [reply Scripts/01_assemble_data_Q2.R:71]: Without the recursive parameter, the call doesn't remove the unzipped stuff on my machine. I think it's likely a naming issue? The zip I downloaded is named HFR_DATIM_FY20Q2.

File: Scripts/02_datim_flags_Q2.R [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      mutate(indicator = "TX_MMD",
+             mer_targets = mer_results,
+             mer_results = NA)
+    
+    df_datim <- bind_rows(df_datim, df_datim_mmd)
+    rm(df_datim_mmd)
+    
+    
+  # remove files (just keep zipped folder)
+    list.files(datim_folder, "3.1 DATIM", full.names = T) %>% unlink(., recursive = TRUE)
```
[the remark this answers is not in the document]
Comment (tessam30) [reply Scripts/02_datim_flags_Q2.R:61]: Sorry, and likely same as above due to naming conventions and folder setup.

File: Scripts/03_assess_completeness_Q2.R [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      mutate(date = as_date(date)) %>% 
+      filter(!(indicator != "TX_CURR" & date >= "2019-12-30")) 
+    
+    #check
+      # df_joint %>%
+      #   count(date, indicator) %>%
+      #   spread(indicator, n)
+
+  #reshape long to preserve NAs for non reporting
+  #TODO: @acheftz - if we are preserving NAs why are they removed in gather?
```
[the remark this answers is not in the document]
Comment (tessam30) [reply Scripts/03_assess_completeness_Q2.R:49]: Sounds good.

File: Scripts/03_assess_completeness_Q2.R [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+# MUNGE -------------------------------------------------------------------
+
+  #change hfr reporting variable for clarity
+    df_joint <- rename(df_joint, hfr_results = val)
+    
+#TODO: @ache - what are doing here for Q2?
+#remove data post Q1 for all but TX_CURR
+    df_joint <- df_joint %>% 
+      mutate(date = as_date(date)) %>% 
+      filter(!(indicator != "TX_CURR" & date >= "2019-12-30"))
```
[the remark this answers is not in the document]
Comment (tessam30) [reply Scripts/03_assess_completeness_Q2.R]: sounds good

**Pull request merged**